### DISCUSSION TITLE
help Torch not compiled with CUDA enabled

### DISCUSSION THREAD

**Discussion by dpv20** [lllyasviel/Fooocus#1694] [Q&A] [answered]
can anyone tell me wich version of each one are u guys using please? i try alredy 3 version, most recently cuda 12.1 with pytorch 2.1.2+cpu 
im still getting this error when i ran python entry_with_update.py

raise AssertionError("Torch not compiled with CUDA enabled")
AssertionError: Torch not compiled with CUDA enabled

any help would be great

**Comment by jndietz** [comment]
Yeah - if you're using `+cpu` this won't work.  Maybe try

```
pip install torch --index-url https://download.pytorch.org/whl/cu121
```

Maybe at least you'll see a different error and we can go from there.

**Comment by dpv20** [comment, marked as answer]
in the end i manage to do it while uninstalling

i ended up using what i found in a forum 

pip uninstall torch
pip cache purge
pip install torch -f https://download.pytorch.org/whl/torch_stable.html
